### PR TITLE
quote command args with & for printing

### LINKED ISSUE #9256: Run output of "tsh db config --format=cmd <db>" may fail
### Description

**What happened**:
Copy-paste output of`tsh db config --format=cmd <db>`, the command may fail.

**What you expected to happen**:
`tsh db config --format=cmd <db>` output should be usable if just copy pasted.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
```
$ ./tsh db config --format=cmd postgres
/opt/homebrew/bin/psql postgres://postgres@<proxy>:443/test?sslrootcert=<certs.pem>&sslcert=<postgres-x509.pem>&sslkey=<sslkey>&sslmode=verify-full
```
Do notice the postgres command contains `&` so terminal will interpret that to launch a few background commands. The command should be **properly quoted** for use.

### Server Details
- Teleport version (run `teleport version`):
- Server OS (e.g. from `/etc/os-release`):
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):
- Additional details:

### Client Details
- Tsh version (`tsh version`): `Teleport v8.0.0-alpha.1 git: go1.17.2`
- Computer OS (e.g. Linux, macOS, Windows): macos
- Browser version (for UI-related issues):
- Installed via (e.g. apt, yum, brew, website download):
- Additional details:

### Debug Logs
not needed

### PR DESCRIPTION
**Pull request by greedy52**
Fixes:
* #9256

closes #9256

## testing
quoted for psql command
```
$ ./tsh db config --format cmd steve-postgres
/opt/homebrew/bin/psql "postgres://alice@teleport.dev.aws.stevexin.me:443/test?sslrootcert=/Users/stevehuang/.tsh/keys/teleport.dev.aws.stevexin.me/certs.pem&sslcert=/Users/stevehuang/.tsh/keys/teleport.dev.aws.stevexin.me/STeve-db/teleport.dev.aws.stevexin.me/steve-postgres-x509.pem&sslkey=/Users/stevehuang/.tsh/keys/teleport.dev.aws.stevexin.me/STeve&sslmode=verify-full"
```

no change for mysql
```
$ ./tsh db config --format cmd steve-rds
/opt/homebrew/opt/mysql-client/bin/mysql --defaults-group-suffix=_teleport.dev.aws.stevexin.me-steve-rds --user alice --database test
```

### CODE REVIEW

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -2469,3 +2470,15 @@ func handleUnimplementedError(ctx context.Context, perr error, cf CLIConf) error
...
 	return trace.WrapWithMessage(perr, errMsgFormat, pr.ServerVersion, teleport.Version)
 }
+
+// printCommand prints command to standard output.
+func printCommand(output io.Writer, cmd *exec.Cmd) {
+	// Add quotes for arguments with '&' to prevent sending the command to
+	// background in terminal.
+	for i, arg := range cmd.Args {
+		if strings.ContainsRune(arg, '&') {
+			cmd.Args[i] = fmt.Sprintf("%q", arg)
```
Comment (Tener) [inline tool/tsh/tsh.go:2480]: We may want to consider more sophisticated approach to be safe. How about what https://github.com/sergeymakinen/go-quote does?

Comment (greedy52) [reply tool/tsh/tsh.go:2480]: ya, it's definitely a valid concern. Seems quoting behaviour is very inconsistent on windows. I thought about it too but couldn't find a good library. 

https://github.com/sergeymakinen/go-quote seems pretty nice and does what we want. But it has a single maintainer i don't know if we want to depend on that. 

How about
```suggestion
               if strings.ContainsRune(arg, '&') && !strings.ContainsRune(arg, '"') {
			cmd.Args[i] = fmt.Sprintf("%q", arg)
```
so we only quote when it's safe to quote. more complicated case we fall back to original behavior. WDYT

Comment (Tener) [reply tool/tsh/tsh.go:2480]: (Looks like my old comment above wasn't submitted properly, you may not have seen it. Sorry!)

Anyway, I gave it more thought, I would suggest one of these options:
1. ignore Windows support but explicitly comment on that deficiency, focus on Linux/Mac support  
2. go the extra mile and support Windows either by writing proper code ourselves or leaning on a third-party package (requires vetting it).

The first option is definitely the cheapest and maybe good enough. 

In particular, trying to detect already-quoted commands feels dodgy and fudges the requirements about the shape of the incoming command to be printed.

**Review by Tener — COMMENTED** [review]
See my other comment for general feedback.

File: tool/tsh/tsh.go (hunk 1 above)
Comment (greedy52) [reply tool/tsh/tsh.go:2480]: @Tener yes, I was just thinking about treating windows differently. Thanks a lot for ideas. I will close this PR and open a different one once I am ready.

**Pull request closed without merging**